packages/ui/src/ui/pages/query-tracker/Plan/NodeStages.tsx

```diff
@@ -59,7 +59,9 @@ function prepareData(data: Stages, state?: NodeState, finishedAt?: string) {
     const stages: StageRow[] = [];
     let [currentStage, currentTime] = Object.entries(data[0])[0];
     const startedTime = currentTime;
-    for (let i = 1; i < data.length; i++) {
+    const dataLength = Object.keys(data).length;
+
+    for (let i = 1; i < dataLength; i++) {
```

Comment:
Are the keys in response might me unordered? If not then it is better to iterate through `Object.keys(data)`:
```ts
Object.keys(data).forEach(() => {
    if (i === "0") {
        return;
    }
   ...
})
```
At the moment it is hard to guess type of `data` looking at the code (object or array).

Reply:
can we change the format of the incoming data?
`
    [...,{name: 'Running', time: '2024-02-19T16:22:55.344038Z'}, ...]
`
the current data format is not the most convenient
Object.keys/entries have bigO equal to O(n) which is much larger than Array.length = O(1)

Reply:
If you are worried about perfomance and you are sure that there are no missing keys in `data` then you can do the next thing:

`for (let i = 1; data[i]; i++) {`